### ISSUE TITLE
include_type_name=true is added to ES 5.x queries eventhough it doesn't support that

### ISSUE THREAD

**Issue by adriancole** [elastic/elasticsearch-hadoop#1276]
It seems the code that adds `include_type_name=true` should guard on versions of ES that support it. Right now, it dies on ES 5.x due to this. I'm not sure which version after 6.3.2 broke this.

```
org.elasticsearch.hadoop.rest.EsHadoopInvalidRequest: org.elasticsearch.hadoop.rest.EsHadoopRemoteException: illegal_argument_exception: request [/test_zipkin:span-2019-04-13/_mapping/span] contains unrecognized parameter: [include_type_name]
null

	at org.elasticsearch.hadoop.rest.RestClient.checkResponse(RestClient.java:465)
	at org.elasticsearch.hadoop.rest.RestClient.execute(RestClient.java:422)
	at org.elasticsearch.hadoop.rest.RestClient.execute(RestClient.java:384)
	at org.elasticsearch.hadoop.rest.RestClient.execute(RestClient.java:388)
	at org.elasticsearch.hadoop.rest.RestClient.get(RestClient.java:168)
	at org.elasticsearch.hadoop.rest.RestClient.getMappings(RestClient.java:315)
	at org.elasticsearch.hadoop.rest.RestClient.getMappings(RestClient.java:305)
	at org.elasticsearch.hadoop.rest.RestRepository.getMappings(RestRepository.java:293)
	at org.elasticsearch.hadoop.rest.RestService.findPartitions(RestService.java:252)
```

```bash
$ curl http://localhost:32801/
{
  "name" : "lY1n3bH",
  "cluster_name" : "elasticsearch",
  "cluster_uuid" : "PDUcXXSeQf-RrFawFZPqzQ",
  "version" : {
    "number" : "5.6.16",
    "build_hash" : "3a740d1",
    "build_date" : "2019-03-13T15:33:36.565Z",
    "build_snapshot" : false,
    "lucene_version" : "6.6.1"
  },
  "tagline" : "You Know, for Search"
}

$ curl http://localhost:32801/test_zipkin-2019-04-13/_mapping/span?include_type_name=true
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"request [/test_zipkin-2019-04-13/_mapping/span] contains unrecognized parameter: [include_type_name]"}],"type":"illegal_argument_exception","reason":"request [/test_zipkin-2019-04-13/_mapping/span] contains unrecognized parameter: [include_type_name]"},"status":400}
```

**Comment by adriancole** [comment]
looks like this happened in the 7.0 beta1 version forward https://github.com/elastic/elasticsearch-hadoop/commit/047d80bc8fb25fdeef059078f4c225d4fe3041bf

**Comment by jbaiera** [comment]
Thanks for reporting, definitely a bug.

**Comment by adriancole** [comment]
was this fixed in the release cut yesterday?

**Comment by jbaiera** [comment]
@adriancole Apologies, I was away on a short break while that release was cut, but should have a patch landing to fix the 5.x compatibility issues soon. The 2.x issues hopefully will be resolved shortly after.

**Comment by jbaiera** [comment]
This should be fixed with #1284
